### PR TITLE
tests

### PR DESCRIPTION
**Pull request by SergKukh**
created user.controller & job.controller test suites
created tests for dto
created user.service & job.service test suites
created e2e test for User & Job controllers

image

image

### CODE REVIEW

**Review by KashyretsIvanna — APPROVED** [review]

File: src/modules/jobs/index.spec.ts [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
+      jobRepository,
+      requestRepository,
+      conversationRepository,
+    );
     jobsController = new JobsController(jobsService);
   });
 
-  describe('getOneJob', () => {
+  describe('getJobById', () => {
     it('should return job by id', async (): Promise<void> => {
```
Comment (lyudad) [inline src/modules/jobs/index.spec.ts]: at least one more test case came to my mind, what if id not exist

File: src/modules/user/user.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    userController = new UserController(userService);
+
+    reqUser = {
+      user: {
+        id: 1,
+      } as User,
+    };
+  });
+
+  it('should be defined', () => {
```
Comment (lyudad) [inline src/modules/user/user.controller.spec.ts]: who defined?

File: src/modules/jobs/index.spec.ts (hunk 1 above)
Comment (SergKukh) [reply src/modules/jobs/index.spec.ts]: @lyudad id is validating by 'class-validator' before calling the controller. So, I created get-job-info-params-dto.spec.ts for testing dto

File: src/modules/user/user.controller.spec.ts (hunk 2 above)
Comment (SergKukh) [reply src/modules/user/user.controller.spec.ts]: @lyudad fixed

File: src/modules/user/dto/add-user-info.dto.spec.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    const errors = await validate(plainToClass(AddUserInfoDto, dto));
+    expect(errors.length).not.toBe(0);
+  });
+
+  it('wrong skill type inside array: should be an error', async (): Promise<void> => {
+    const dto = { skills: ['1'] };
+    const errors = await validate(plainToClass(AddUserInfoDto, dto));
+    expect(errors.length).not.toBe(0);
+  });
+});
```
Comment (lyudad) [inline src/modules/user/dto/add-user-info.dto.spec.ts:66]: its just dto, what about endpoints, services?

Comment (SergKukh) [reply src/modules/user/dto/add-user-info.dto.spec.ts:66]: @lyudad added user.service & job.service test suites, also e2e tests for user & job controllers

File: src/modules/jobs/dto/get-job-info-params-dto.spec.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import GetJobInfoParamsDto from './get-job-info-params-dto';
+
+describe('GetJobInfoParamsDto', () => {
+  it('should be validated successfully', async (): Promise<void> => {
+    const dto = { id: '1' };
+    const errors = await validate(plainToClass(GetJobInfoParamsDto, dto));
+    expect(errors.length).toBe(0);
+  });
+
+  it('id does not exist: should be an error', async (): Promise<void> => {
```
Comment (lyudad) [inline src/modules/jobs/dto/get-job-info-params-dto.spec.ts]: id of what? job id, user id

Comment (SergKukh) [reply src/modules/jobs/dto/get-job-info-params-dto.spec.ts]: @lyudad fixed

**Pull request merged**